### PR TITLE
Clean/expand TODO's to use org capabilities

### PR DESCRIPTION
**Pull request by zachmandeville**
**This PR depends on PR #3**

This PR expands on #3, this time to make the best use of (magit-list-todos).  This function lets us type `spc p t` from within a git repo and see all TODO created in that repo.  Our repository has a lot of noise in regards to TODO, though, which lessened the effectiveness of this command.  This PR cleans up the repo to reduce the noise.  I also wanted to remove now completed TODO's from files, but this required expanding on how we completed it (specifically in our incident report). So this PR also includes an expanded write-up in our incident report.

Main Changes in this PR:
- consolidate all TODO's into our todo.org file
- add .dir-locals.el file to the repo that tells it to ignore any todo's found in yaml files (as our upstream yaml's contain a lot of dummy todo's)
- write up a summary of our work debugging mattermost.

### CODE REVIEW

**Comment by riaankleinhans** [comment]
LGTM

File: .dir-locals.el [hunk 1]
```diff
@@ -0,0 +1,4 @@
+;;; Directory Local Variables
+;;; For more information see (info "(emacs) Directory Variables")
+
+((magit-status-mode . ((magit-todos-exclude-globs . ("*.yaml")))))
```
Comment (BobyMCbobs) [inline .dir-locals.el:4]: This means to ignore any YAMl file changes in the current directory in magit?

File: mattermost/incident-reports/2022-07-07_install-mattermost/README.org [hunk 2]
```diff
@@ -1,35 +1,108 @@
...
 #+PROPERTY: header-args: :comment no
 
-* Background
-On 4 July 2022, we attempted an upgrade to mattermost, going from 5.37 to 7.0.
-We did this on our production environment, thinking it would be a simple change
-in our mattermost manifest with an hour of downtime. There were errors in the
-upgrade, and when we tried to downgrade back to 5.37, we were unable to connect
-mattermost to the database. Our chat infrastructure was down, and it was unclear
-how we would get it up.
+* Summary
```
Comment (BobyMCbobs) [inline mattermost/incident-reports/2022-07-07_install-mattermost/README.org:4]: :+1:

**Comment by BobyMCbobs** [comment]
This is much improved!

**Pull request merged**